lib/cucumber/formatter/legacy_api/adapter.rb

```diff
@@ -54,7 +54,7 @@ def printer
         end
 
         def record_test_case_result(test_case, result)
-          scenario = LegacyResultBuilder.new(result).scenario(test_case.name, test_case.location)
+          scenario = LegacyResultBuilder.new(result).scenario("#{test_case.keyword}: #{test_case.name}", test_case.location)
```

Comment:
Is there a better place to combine these?  It doesn't seem that nice to format that string in the argument to `#scenario`.  Would it be better to pass the whole `test_case`?  The formatting in the `#name` method in facade.rb seems ok.